pkg/gator/sync/test/test_test.go

```diff
@@ -0,0 +1,205 @@
+package test
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/cachemanager/parser"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/gator/fixtures"
+	"github.com/open-policy-agent/gatekeeper/v3/pkg/gator/reader"
+	"github.com/stretchr/testify/require"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+func TestTest(t *testing.T) {
+	tcs := []struct {
+		name         string
+		inputs       []string
+		omitManifest bool
+		wantReqs     map[string]parser.SyncRequirements
+		wantErrs     map[string]error
+		err          error
+	}{
+		// {
+		// 	name: "basic req unfulfilled",
+		// 	inputs: []string{
+		// 		fixtures.TemplateReferential,
+		// 	},
+		// 	omitManifest: true,
+		// 	wantReqs: map[string]parser.SyncRequirements{
+		// 		"k8suniqueserviceselector": {
+		// 			parser.GVKEquivalenceSet{
+		// 				{
+		// 					Group:   "",
+		// 					Version: "v1",
+		// 					Kind:    "Service",
+		// 				}: struct{}{},
+		// 			},
+		// 		},
+		// 	},
+		// 	wantErrs: map[string]error{},
+		// },
+		// {
+		// 	name: "one template having error stops requirement evaluation",
+		// 	inputs: []string{
+		// 		fixtures.TemplateReferential,
+		// 		fixtures.TemplateReferentialBadAnnotation,
+		// 	},
+		// 	omitManifest: true,
+		// 	wantReqs:     nil,
+		// 	wantErrs: map[string]error{
+		// 		"k8suniqueingresshostbadannotation": fmt.Errorf("json: cannot unmarshal object into Go value of type parser.CompactSyncRequirements"),
+		// 	},
+		// },
+		// {
+		// 	name: "basic req fulfilled by syncset",
+		// 	inputs: []string{
+		// 		fixtures.TemplateReferential,
+		// 		fixtures.Config,
+		// 	},
+		// 	omitManifest: true,
+		// 	wantReqs:     map[string]parser.SyncRequirements{},
+		// 	wantErrs:     map[string]error{},
+		// },
+		{
+			name: "basic req fulfilled by syncset and supported by cluster",
+			inputs: []string{
+				fixtures.TemplateReferential,
+				fixtures.Config,
+				fixtures.GVKManifest,
+			},
+			wantReqs: map[string]parser.SyncRequirements{},
+			wantErrs: map[string]error{},
+		},
+		// {
+		// 	name: "basic req fulfilled by syncset but not supported by cluster",
+		// 	inputs: []string{
+		// 		fixtures.TemplateReferential,
+		// 		fixtures.Config,
+		// 		fixtures.GVKManifest,
```

Comment:
Outside of objects which are hard to write because they have so much content (templates and constraints), I think we should move away from using fixtures.

When I look at this test, I have to go look elsewhere to figure out what the meaningful contents of these objects is.  That makes it hard for me to look at the test and know if it's right.  

If you'd like to create helper functions to make it easy to generate these objects, I recommend putting them in the [fakes](https://github.com/open-policy-agent/gatekeeper/tree/master/pkg/fakes).  There are examples there that you could use in creating new functions.

Potential helper functions:

```
func SyncSet(gvks []scheme.GroupVersionKind) SyncSet {

func GVKManifest(gvks []scheme.GroupVersionKind) GVKManifest {
```

Calling these functions in your test case allows you to declare the contents of the object (the meaningful part for the test) without including other boilerplate in the test case.